frameworks/Python/apidaora/app.py

```diff
@@ -0,0 +1,202 @@
+    return query_count
+
+
+connection_pool = None
+sort_fortunes_key = itemgetter(1)
+template = load_fortunes_template()
+loop = asyncio.get_event_loop()
+loop.run_until_complete(setup_database())
+
+
+@jsondaora
+class JsonBody(TypedDict):
+    message: str
+
+
+@jsondaora
+class JsonResponse(JSONResponse):
+    body: JsonBody
+
+
+async def json_serialization(req: Request) -> JsonResponse:
+    return JsonResponse(
+        status_code=HTTPStatus.OK,
+        body=JsonBody(message='Hello, world!')
+    )
+
+
+@jsondaora
+class SingleDatabaseBody(TypedDict):
+    id: int
+    randomNumber: float
+
+
+@jsondaora
+class SingleDatabaseResponse(JSONResponse):
+    body: SingleDatabaseBody
+
+
+async def single_database_query(req: Request) -> SingleDatabaseResponse:
+    row_id = randint(1, 10000)
+
+    async with connection_pool.acquire() as connection:
+        number = await connection.fetchval(READ_ROW_SQL, row_id)
+
+    return SingleDatabaseResponse(
+        status_code=HTTPStatus.OK,
+        body=SingleDatabaseBody(id=row_id, randomNumber=number)
+    )
+
+
+@jsondaora
+class MultipleDatabaseRequestQuery(TypedDict):
+    queries: Optional[str] = None
+
+
+@jsondaora
+class MultipleDatabaseRequest(Request):
+    query: MultipleDatabaseRequestQuery
+
+
+@jsondaora
+class MultipleDatabaseObject(TypedDict):
+    id: int
+    randomNumber: float
+
+
+@jsondaora
+class MultipleDatabaseBody(TypedDict):
+    id: int
+    randomNumber: float
+
+
+@jsondaora
+class MultipleDatabaseResponse(JSONResponse):
+    body: List[MultipleDatabaseBody]
+
+
+async def multiple_database_queries(
+    req: MultipleDatabaseRequest
+) -> MultipleDatabaseResponse:
+    num_queries = get_num_queries(req.query['queries'])
+    row_ids = [randint(1, 10000) for _ in range(num_queries)]
+    worlds = []
+
+    async with connection_pool.acquire() as connection:
+        statement = await connection.prepare(READ_ROW_SQL)
+        for row_id in row_ids:
+            number = await statement.fetchval(row_id)
+            worlds.append(
+                MultipleDatabaseObject(
+                    id=row_id,
+                    randomNumber=number
+                )
+            )
+
+    return MultipleDatabaseResponse(status_code=HTTPStatus.OK, body=worlds)
+
+
+@jsondaora
+class FortunesResponse(HTMLResponse):
+    body: str
+
+
+async def fortunes(req: Request) -> FortunesResponse:
+    async with connection_pool.acquire() as connection:
+        fortunes = await connection.fetch('SELECT * FROM Fortune')
+
+    fortunes.append(ADDITIONAL_ROW)
+    fortunes.sort(key=sort_fortunes_key)
+    content = template.render(fortunes=fortunes)
+    return FortunesResponse(
+        status_code=HTTPStatus.OK,
+        body=content
+    )
+
+
+async def database_updates(
+    req: MultipleDatabaseRequest
+) -> MultipleDatabaseResponse:
+    num_queries = get_num_queries(req.query['queries'])
+    updates = [(randint(1, 10000), randint(1, 10000)) for _ in range(num_queries)]
```

Comment:
The updates test requires that you query `num_queries` world objects from the database first, then update that world object's random number field. If I'm reading this correctly, you're just creating random world objects without querying them here.

Reply:
@nbrady-techempower I will update the test. Thanks for replying!

I copied the tests from fastapi, so I think the fastapi tests are wrong too.

Reply:
the code queries the results first

```
await stmt.fetchval(id_)
```

Reply:
Hmm, ok. I'm going to get back to you on this shortly and look at some of the other tests. There may have been a discussion about this already but I don't remember. I think other python tests will need to be fixed. The data should be read from the database, mapped to an object,  and then have its `randomNumber` field updated. In this implementation, `statement.fetchval` could do literally anything because the result is never read.

Reply:
OK! Now I understand your point @nbrady-techempower 

I will update my test to set the retrived values on objects, update its values with random numbers, then update the database values.

when my fixed tests is OK, I can updated others python tests too in other PR

Reply:
Thanks! That'd be great!

Reply:
@nbrady-techempower
I updated the tests, now I think it does what you describe.